install.sh

```diff
@@ -0,0 +1,276 @@
+#==================================#
+
+# Make sure script dependencies are installed on bare bones installations
+instlog "Installing install script dependencies"
+sudo apt-get update -qq
+sudo apt-get install -qq wget ssh git lsb-release fakeroot
+
+# Add software repositories for ROS and Gazebo
+instlog "Adding ROS and Gazebo PPAs to software sources"
+sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu trusty main\" > /etc/apt/sources.list.d/ros-latest.list"
+sudo sh -c "echo \"deb http://packages.osrfoundation.org/gazebo/ubuntu trusty main\" > /etc/apt/sources.list.d/gazebo-latest.list"
+
+# Get the GPG signing keys for the above repositories
+wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
+sudo apt-key adv --keyserver hkp://pool.sks-keyservers.net --recv-key 0xB01FA116
+
+# Install ROS and other project dependencies
+instlog "Installing ROS and Gazebo"
+sudo apt-get update -qq
+sudo apt-get install -qq python-catkin-pkg python-rosdep ros-indigo-desktop
+sudo apt-get install -qq gazebo7
+
+# Source ROS configurations for bash on this user account
+source /opt/ros/indigo/setup.bash
+if !(cat ~/.bashrc | grep --quiet "source /opt/ros"); then
+	echo "" >> ~/.bashrc
+	echo "# Sets up the shell environment for ROS" >> ~/.bashrc
+	echo "source /opt/ros/indigo/setup.bash" >> ~/.bashrc
+fi
+
+# Get information about ROS versions
+instlog "Initializing ROS"
+if !([ -f /etc/ros/rosdep/sources.list.d/20-default.list ]); then
+	sudo rosdep init > /dev/null 2>&1
+fi
+rosdep update
+
+
+#=================================#
+# Workspace and Repository Set Up #
+#=================================#
+
+# Set up catkin workspace directory
+if !([ -f $CATKIN_DIR/src/CMakeLists.txt ]); then
+	instlog "Generating catkin workspace at $CATKIN_DIR"
+	mkdir -p "$CATKIN_DIR/src"
+	cd "$CATKIN_DIR/src"
+	catkin_init_workspace
+	catkin_make -C "$CATKIN_DIR"
+else
+	instlog "Using existing catkin workspace at $CATKIN_DIR"
+fi
+
+# Source the workspace's configurations for bash on this user account
+source "$CATKIN_DIR/devel/setup.bash"
+if !(cat ~/.bashrc | grep --quiet "source $CATKIN_DIR/devel/setup.bash"); then
+	echo "source $CATKIN_DIR/devel/setup.bash" >> ~/.bashrc
+fi
+
+# Check if the Navigator repository is present; if it isn't, download it
+if !(ls "$CATKIN_DIR/src" | grep --quiet "Navigator"); then
+	instlog "Downloading up the Navigator repository"
+	cd $CATKIN_DIR/src
+	git clone -q https://github.com/uf-mil/Navigator.git
+	cd $CATKIN_DIR/src/Navigator
+	git remote rename origin upstream
+	instlog "Make sure you change your git to point to your own fork! (git remote add origin your_forks_url)"
+fi
+
+
+#===================================#
+# Navigator Dependency Installation #
+#===================================#
+
+instlog "Installing Navigator dependencies from the Ubuntu repositories"
+sudo apt-get install -qq libompl-dev
+sudo apt-get install -qq python-pygame
+
+instlog "Installing Navigator ROS dependencies"
```

Comment:
Ideally there should be two install scripts, or an environment variable to determine what type of machine we're on. For semaphore ros-indigo-desktop is fine, but everyone should be running ros-indigo-desktop-full on their machine. This would remove our issues with having to enumerate every single package. Do we have any idea if the time hit with installing desktop-full on semaphore is big enough to make desktop more appealing?